hircluster.c

```diff
@@ -2398,8 +2381,6 @@ static void *redis_cluster_command_execute(redisClusterContext *cc,
 
     c = ctx_get_by_node(cc, node);
     if (c == NULL) {
-        return NULL;
-    } else if (c->err) {
         node = node_get_which_connected(cc);
```

Comment:
I'm thinking maybe there is a point to distinguish between initial connect failure and reconnect failure. Maybe that was the idea of the old code. I hope you can prove me wrong. :-)

What if connection fails the first time when trying to connect to a node? We will pick a different node, get a MOVED redirect, update the slot mapping, then try to connect the failing node again and loop(?).

The idea of doing it only at failed *re*connect is that hopefully an updated slot mapping will give us the right node. It may still loop though.

Reply:
Looks like this has been there from day 1 in hiredis-vip, so git blame didn't help :)
If its a scaledown then the node can be gone before we attempt to connect to for the first time, then we need to get a new topology view.
But for a problem like a routing problem, then we will do max-retry-count number of attempts (default 5) and getting the same topology each time, a bit wasteful maybe?
There is currently no check that a topology view update via `cluster_update_route()` gives the same view as we already know, maybe that would help for that case..

Reply:
The whole random node thing is a quite dangerous. What if we want to send an admin command to a specific node (using ToNode) which is down. Then it is sent to a random node instead(!).

I think we can update the slotmap, then try again to fetch the node responsible for a certain key, but not send a command to a random node.